src/components/PurposeEditDialog/helpers.js

```diff
@@ -0,0 +1,32 @@
+import set from 'lodash/set'
+import cloneDeep from 'lodash/cloneDeep'
+
+export const createGeojsonWithModifiedPurpose = ({
+  geojson,
+  tripId,
+  purpose
+}) => {
+  const matchedTrip = geojson.series
+    .map((serie, index) => {
+      if (serie.id === tripId) {
+        return {
+          serie,
+          index
+        }
+      }
+    })
+    .find(e => e)
```

Comment:
nit: Maybe this would be more concise/readable with findIndex (very subjective opinion).

```js
const index = geojson.series.findIndex(serie => serie.id === tripId)
if (index > -1) {
   const serie = geojson.series[index]
}
```